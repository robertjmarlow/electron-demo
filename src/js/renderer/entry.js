const { ipcRenderer } = require('electron');
const $ = require('jquery');
const os = require('os');
require('../../css/main.css');

$(document).ready(function() {
  ipcRenderer.on('getFileContents', function(event, arg) {
    $('#file-contents').append('<div class="fileContents">' + arg.toString().replace(/\n/g, '<br/>').replace(/ /g, '&nbsp') + '</div>');
  });

  ipcRenderer.on('getDirectoryContents', function(event, arg) {
    // yes, this was broken in the last checkin...
    $('#dir-contents').append('<div class="fileList">' + arg + '</div>');
  });

  ipcRenderer.send('getFileContents', './main.js');
  ipcRenderer.send('getDirectoryContents', '.');

  $('#os-stuff')
    .append('<div>Your OS is: <span class="big">' + os.platform() + ' ' + os.release() + '</span></div>')
    .append('<div>Your uptime (in seconds) is: <span class="big">' + os.uptime() + '</span></div>')
});
