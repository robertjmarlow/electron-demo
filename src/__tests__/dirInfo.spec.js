import React from 'react';
import renderer from 'react-test-renderer';
import DirInfo from '../jsx/DirInfo';

jest.mock('electron');

describe('Directory Info', () => {
  it('shows directory contents', () => {
    const tree = renderer.create(
      <DirInfo/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
