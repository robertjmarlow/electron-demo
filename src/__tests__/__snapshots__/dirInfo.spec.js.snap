exports[`Directory Info shows directory contents 1`] = `
<div
  className="dirContents">
  <div>
    .
  </div>
  <div>
    ..
  </div>
  <div>
    foo
  </div>
  <div>
    bar
  </div>
  <div>
    baz
  </div>
</div>
`;
